### ISSUE TITLE
wny does the numpy.ma (masked) array mean method have a "special case" return type

### ISSUE THREAD

**Issue by dermen** [numpy/numpy#7833]
I noticed that the [numpy masked-array mean method](http://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.MaskedArray.mean.html#numpy.ma.MaskedArray.mean) returns different types when it probably should not:

``` python
import sys
sys.version
#'2.7.9 (v2.7.9:648dcafa7e5f, Dec 10 2014, 10:10:46) \n[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]'
import numpy as np
np.__version__
#'1.9.2'

A = np.ma.masked_equal([1,1,0], value=0)
B = np.ma.masked_equal([1,1,1], value=0) # no masked values

type(A.mean())
#numpy.float64
type(B.mean())
#numpy.ma.core.MaskedArray
```

Other `numpy.ma.core.MaskedArray` methods seem to be consistent

``` python
type( A.sum()) == type(B.sum())
# True
type( A.prod()) == type(B.prod())
# True
type( A.std()) == type(B.std())
# True
type( A.mean()) == type(B.mean())
# False
```

This kind of behavior seems undesirable.

It is worth pointing out that

``` python
C = np.ma.masked_array([1, 1, 1], mask=[False, False, False])
type(C.mean()) == type(A.mean())
# True 
```

You can see my original stack overflow question [here](http://stackoverflow.com/questions/38164773/numpy-ma-masked-array-mean-method-has-inconsitent-return-type)

**Comment by eric-wieser** [comment]
This is fixed at least as early as 1.12.1 - both give `np.float64`

**Comment by eric-wieser** [comment]
Duplicate of #5769
